test/e2e/provider_expectations.go

```diff
@@ -298,3 +306,316 @@ func (tc *testConfig) ExpectNodeToBeDrainedBeforeDeletingMachine() error {
 		return true, nil
 	})
 }
+
+var nodeDrainLabels = map[string]string{
+	workerRoleLabel:      "",
+	"node-draining-test": "",
+}
+
+func machineFromMachineset(machineset *mapiv1beta1.MachineSet) *mapiv1beta1.Machine {
```

Comment:
I suspect you are doing this process because machineSet does not reconcile labels to machines today, so it'd be nice to add a comment/TODO so eventually we can add another test case for just scaling down a set

Reply:
It's done to get a machine template. This way I don't need to assume anything about machine provider config. I just take what is already provided by the cluster. In case of openshift installer I don't have a knowledge of which AMI I should use so I take one that is rendered by the installer.